From 0180cbf322c39a7e7176c2ae9d6bb89b83a9b235 Mon Sep 17 00:00:00 2001
From: Etienne Carriere <etienne.carriere@foss.st.com>
Date: Mon, 15 Apr 2024 10:52:25 +0200
Subject: [PATCH 0463/1695] scpfw: module: scmi_clock: 64bit clock rate
 computation

Correct variable type used are scmi_clock_round_rate_get_handler()
argument reference as reported by compilation toolchain message:

module/scmi_clock/src/mod_scmi_clock.c: In function 'scmi_clock_round_rate_get_handler':
module/scmi_clock/src/mod_scmi_clock.c:1596:63: error: passing argument 3 of 'scmi_clock_ctx.clock_api->round_rate' from incompatible pointer type [-Werror=incompatible-pointer-types]
 1596 |                                                   input_rate, &output_rate);
      |                                                               ^~~~~~~~~~~~
      |                                                               |
      |                                                               long unsigned int *
module/scmi_clock/src/mod_scmi_clock.c:1596:63: note: expected 'uint64_t *' {aka 'long long unsigned int *'} but argument is of type 'long unsigned int *'

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/optee_os.git 86152f0d9503bb6763977eda8ccb0adc26ada399 scpfw: module: scmi_clock: 64bit clock rate computation]
Change-Id: Icbd7a2cb357d1442342e40d1e6cd2bb65f1b20a5
Signed-off-by: Etienne Carriere <etienne.carriere@foss.st.com>
---
 .../SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c         | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/core/lib/scmi-server/SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c b/core/lib/scmi-server/SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c
index 330ce7359..5ee7a38f4 100644
--- a/core/lib/scmi-server/SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c
+++ b/core/lib/scmi-server/SCP-firmware/module/scmi_clock/src/mod_scmi_clock.c
@@ -1553,7 +1553,7 @@ static int scmi_clock_round_rate_get_handler(fwk_id_t service_id,
     };
     const struct mod_scmi_clock_device *clock_device;
     unsigned long input_rate;
-    unsigned long output_rate;
+    uint64_t output_rate;
     size_t response_size;
     int status;
 
-- 
2.47.2

